### PR TITLE
feat: Add weight cut reduction to GSF

### PR DESCRIPTION
**Pull request by benjaminhuth**
A faster but less accurate mixture reduction method.

### CODE REVIEW

File: Core/src/TrackFitting/GsfMixtureReduction.cpp [hunk 1]
```diff
@@ -49,6 +49,18 @@ void reduceWithKLDistanceImpl(std::vector<Acts::GsfComponent> &cmpCache,
...
 
+void reduceMixtureLargestWeights(std::vector<GsfComponent> &cmpCache,
+                                 std::size_t maxCmpsAfterMerge,
+                                 const Surface &) {
+  if (cmpCache.size() <= maxCmpsAfterMerge) {
+    return;
+  }
+
+  std::sort(cmpCache.begin(), cmpCache.end(),
+            [](const auto &a, const auto &b) { return a.weight > b.weight; });
```
Comment (andiwand) [inline Core/src/TrackFitting/GsfMixtureReduction.cpp:60]: I wonder if `nth_element` would be more optimal

no need to change

File: docs/_extensions/lazy_autodoc.py [hunk 2]
```diff
@@ -153,7 +155,7 @@ def process_roles(file: Path) -> List[Tuple[str, str]]:
     # add members to their parents
 
     api_preamble = """
-    """
+"""
```
Comment (andiwand) [inline docs/_extensions/lazy_autodoc.py:158]: intended change?

**Review by andiwand — APPROVED** [review]

File: docs/_extensions/lazy_autodoc.py (hunk 2 above)
Comment (benjaminhuth) [reply docs/_extensions/lazy_autodoc.py:158]: Indeed! This fixes incorrect rendering in the API documentation (see the heading `## Classes` in https://acts.readthedocs.io/en/latest/api/api.html)

**Review by andiwand — DISMISSED** [review]

**Comment by asalzburger** [comment]
@benjaminhuth - we have an unused variable here, can you fix it?

**Comment by benjaminhuth** [comment]
@asalzburger fixed, can you approve?

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [239b9420dab9f89b20e614b8a9149301180c74d9]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6484735)!

|status|job|
|------|---|
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33975781) |
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973804) |
|:red_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973803) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973802) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973801) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973800) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973799) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973798) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973797) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973796) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973795) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973794) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973793) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973792) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973791) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973790) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973788) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33973787) |

**Pull request merged**